pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -1234,6 +1234,9 @@ protected void handleBrokerRegistrationEvent(String broker, NotificationType typ
             handleBrokerCreationEvent(broker);
         } else if (type == NotificationType.Deleted) {
             log.info("BrokerRegistry detected the broker:{} registry has been deleted.", broker);
+            if (brokerRegistry.getBrokerId().equals(broker)) {
+                brokerRegistry.registerAsync();
```

Comment:
Nit: It is counter-intuitive that this broker is still alive but receives deletion events(maybe network issue?).

Can you add a comment for this case?

Reply:
Added.